docs/proposal-multi-user-approach.md

```diff
@@ -13,11 +12,276 @@ via a UI Server running as the suite owner for UI purposes meaning that, from
 applicable, made accessible to the ``cylc`` CLI tools via the GraphQL
 interface.
+
+Work discussed here will be implementing accessing of other users' UI Servers, the orange arrow in the diagram below.
+
+![Cylc-8 Authorisation](img/cylc-8-auth.png "Cylc-8 Authorisation")
+
+## Security Considerations
+
+In line with the guidance provided by [section 4.1 of the OWASP ASVS](https://github.com/OWASP/ASVS/blob/master/4.0/en/0x12-V4-Access-Control.md#v41-general-access-control-design) (Application Security Verification Standard):
+
+* Authorisation will be *fail secure* and *deny by default*. That is, unless explicit authorisation has been granted to the user to perform a given action, UI Servers will be locked down.
+* The *principle of least privilege* should be adhered to, whereby users are only granted those privileges that are essential to the work they are doing. To achieve this, UI Server owners will need a sufficient level of granularity for configuring access.
+
+UI Server back-end code should prevent users manipulating client-side code to bypass authorisation. For example, sending a GraphQL query for a different operation to that stated in the operation name parameter.
+
+## Configuration
+
+Two configuration files will be needed, one for site and one for users. This authorisation configuration can be added to the existing UI Server config files, although this may prove a technical challenge. Authorisation configurations should be loaded at start up and stored as a dictionary. This means any changes to config will require the UI Server to be restarted before they are applied.
+
+### Config Design
+
+The config discussed at [CylcCon2020](https://github.com/cylc/cylc-uiserver/issues/10#issuecomment-773752542) suggests admin levels of `read`, `write` and `execute`. Reducing the number of access groups is another suggestion, especially since write and execute are not very meaningful to users (in terms of Cylc) and to ensure users do not accidentally give away more privileges than intended, a simpler system, as outlined below, may be preferential. Proposal: `READ` and `ADMIN` permission groups.
+
+### READ and ADMIN
+
+For the purposes of clarity for users, and since we have operation (mutation) level granularity, the simpler method of having two groups, `READ` and `ADMIN`, rather than three. Access groups should be entered in caps to avoid confusing e.g. READ with read mutation.
+
+### Example UI Server User Configuration
+
+```python
+c.UIServer.authorisation = {
+    "<user1>": ["READ", "pause", "trigger", "message"], # Specified interactions allowed
+    "group:<group1>": ["ADMIN"],                # All access to workflows granted
+    "<user2>": ["ADMIN", "!trigger", "!edit"],  # All Admin, not trigger, or edit
+    "<user3>": ["!ADMIN"]                       # No privileges (incl READ)
+}
+```
+
+### Example Site Configuration
+
+```python
+c.UIServer.authorisation = {
```

Comment:
I think you'll need to use a different name to distinguish this from the user config.